game.cpp

```diff
@@ -11,7 +11,7 @@
 #include <iostream>
 #include <random>
 
-game::game(double wall_short_side,
+game::game(environment the_environment,
```

Comment:
Nice, a new feature! What would be even better, though, is to use `const& environment`, see [the C++ core guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-in).